openpype/hosts/maya/plugins/publish/validate_resolution.py

```diff
@@ -0,0 +1,99 @@
+import pyblish.api
+from openpype.pipeline import (
+    PublishValidationError,
+    OptionalPyblishPluginMixin
+)
+from maya import cmds
+from openpype.pipeline.publish import RepairAction
+from openpype.hosts.maya.api import lib
+from openpype.hosts.maya.api.lib import reset_scene_resolution
+
+
+class ValidateSceneResolution(pyblish.api.InstancePlugin,
+                              OptionalPyblishPluginMixin):
+    """Validate the render resolution setting aligned with DB"""
+
+    order = pyblish.api.ValidatorOrder - 0.01
+    families = ["renderlayer"]
+    hosts = ["maya"]
+    label = "Validate Resolution"
+    actions = [RepairAction]
+    optional = True
+
+    def process(self, instance):
+        if not self.is_active(instance.data):
+            return
+        invalid = self.get_invalid_resolution(instance)
+        if invalid:
+            raise PublishValidationError(
+                "issues occurred", description=(
+                    "Wrong render resolution setting. "
+                    "Please use repair button to fix it.\n"
+                    "If current renderer is V-Ray, "
+                    "make sure vraySettings node has been created"))
+
+    def get_invalid_resolution(self, instance):
+        width, height, pixelAspect = self.get_db_resolution(instance)
+        current_renderer = cmds.getAttr(
+            "defaultRenderGlobals.currentRenderer")
+        layer = instance.data["renderlayer"]
+        invalids = []
+        if current_renderer == "vray":
+            vray_node = "vraySettings"
+            if cmds.objExists(vray_node):
+                current_width = lib.get_attr_in_layer(
+                    "{}.width".format(vray_node), layer=layer)
+                current_height = lib.get_attr_in_layer(
+                    "{}.height".format(vray_node), layer=layer)
+                current_pixelAspect = lib.get_attr_in_layer(
+                    "{}.pixelAspect".format(vray_node), layer=layer
+                )
+            else:
+                invalid = self.log.error(
+                    "Can't detect VRay resolution because there is no node "
+                    "named: `{}`".format(vray_node)
+                )
+                invalids.append(invalid)
```

Comment:
```suggestion
                self.log.error(
                    "Can't detect VRay resolution because there is no node "
                    "named: `{}`".format(vray_node)
                )
                return True
```